obspy/core/utcdatetime.py

```diff
@@ -383,8 +396,36 @@ def __init__(self, *args, **kwargs):
             kwargs['second'] = int(_sec)
             kwargs['microsecond'] = int(round(_frac * 1e6))
             args = args[0:5]
-        dt = datetime.datetime(*args, **kwargs)
-        self._from_datetime(dt)
+
+        try:  # If a value Error is raised try to allow overflow (see #2222)
+            dt = datetime.datetime(*args, **kwargs)
+        except ValueError:
+            if not strict:
+                self._handle_overflow(*args, **kwargs)
+            else:
+                raise
+        else:
+            self._from_datetime(dt)
+
+    def _handle_overflow(self, year, month, day, hour=0, minute=0, second=0,
+                         microsecond=0):
+        """
+        Handles setting date if an overflow of usual value limits is detected.
+        """
+        kwargs = locals()
+        kwargs.pop('self')
+        # ensure hour minute second are in kwargs
+        extra_seconds = 0
+        # keep track of second equiv. of overflow
+        extra_seconds += (kwargs['hour'] // 24) * 3600 * 24
+        extra_seconds += (kwargs['minute'] // 60) * 60 * 60
+        extra_seconds += (kwargs['second'] // 60) * 60
```

Comment:
I don't really like dropping decimal parts silently here. The user should know when they are providing bad/unexpected input.

Since this is basically a drop-in replacement method for `datetime.datetime(*args, **kwargs)`, I would propose to either..

 - be strict with types, i.e. raise `TypeError`s if e.g. `hour` is not an `int` like `datetime` does.. (guess this would be semi strict then)
 - or accept floats but don't silently drop the decimal part like done right now (ultra sloppy mode)

Personally I would go for the first option.

Reply:
Actually the purpose of the int division here is not to drop the decimal parts, but simply to get the second equivalent of overflow (if hour, minute, or second exceed their natural bounds).  Looking at it again, I can see this is  overly complicated, I will simplify it.

Option one is already implemented. This is because the values are given to `datetime.datetime` (line 401) which will raise a `TypeError` if a float for any of those values is passed. Since we are only catching `ValueError` to determine if the overflow logic should kick in, any `TypeError`s will still propagate up the stack.